editoast/src/views/level_crossing_occupancy.rs

```diff
@@ -72,7 +117,520 @@ pub(in crate::views) async fn occupancy(
     })
     .await?;
 
-    //TODO: implement real logic
+    let conn = &mut db_pool.get().await?;
+
+    // Load infra
+    let infra = Infra::retrieve_or_fail(conn.clone(), infra_id, || {
+        LevelCrossingError::InfraNotFound { infra_id }
+    })
+    .await?;
+
+    // Load level crossings
+    let level_crossings: Vec<LevelCrossingModel> =
+        LevelCrossingModel::retrieve_batch_or_fail(conn, level_crossing_ids, |missing| {
+            LevelCrossingError::LevelCrossingBatchNotFound {
+                count: missing.len(),
+            }
+        })
+        .await?;
+
+    // Load trains
+    let trains: Vec<PacedTrain> = PacedTrain::retrieve_batch_or_fail(conn, train_ids, |missing| {
+        LevelCrossingError::TrainBatchNotFound {
+            count: missing.len(),
+        }
+    })
+    .await?;
+
+    // Collect all occurrences from all trains
+    let train_occurrences = trains
+        .iter()
+        .flat_map(|train| {
+            train
+                .iter_occurrences()
+                .map(|(occurrence_id, train_schedule)| (train.id, occurrence_id, train_schedule))
+        })
+        .collect_vec();
+
+    // Extract train schedules for simulation
+    let train_schedules = train_occurrences
+        .iter()
+        .map(|(_, _, train_schedule)| train_schedule.clone())
+        .collect_vec();
+
+    let simulations_result = train_simulation_batch(
+        conn,
+        valkey_client,
+        core_client,
+        &train_schedules,
+        &infra,
+        electrical_profile_set_id,
+        config.app_version.as_deref(),
+    )
+    .await?;
+
+    // Load rolling stocks lengths
```

Comment:
Nit: do we really need this kind of comment? The function called below has the exact same name, except for the word separators.